index.js

```diff
@@ -581,7 +581,7 @@ function parse (args, opts) {
       }
     } else if (o[key] === undefined && isTypeArray) {
       o[key] = isValueArray ? value : [value]
-    } else if (duplicate && !(o[key] === undefined || checkAllAliases(key, flags.bools) || checkAllAliases(keys.join('.'), flags.bools) || checkAllAliases(key, flags.counts))) {
+    } else if (duplicate && !(o[key] === undefined || o[key] === value || checkAllAliases(key, flags.bools) || checkAllAliases(keys.join('.'), flags.bools) || checkAllAliases(key, flags.counts))) {
```

Comment:
this would introduce issues for folks who are trying to create an array like so:

```
parser.js -x 9 -x 9
```

I believe, I think a more general solution would be to use logic similar to this:

https://github.com/yargs/yargs-parser/blob/master/index.js#L352

and introduce [decamelize](https://www.npmjs.com/package/decamelize), so that we handle both going from `foo-bar` to `fooBar` and `fooBar` to `foo-bar`.